### PR TITLE
Add RSS Feed for forum

### LINKED ISSUE #339: RSS Feed
Hi, i checked the previous commit https://github.com/laravelio/portal/search?q=rss&type=Commits&utf8=%E2%9C%93 and found that laravel.io has RSS feed before, but it seems now it can be accessed, not found.  
  
So, is there the RSS feed removed by purpose or missed it after redesign the site?

### PR DESCRIPTION
**Pull request by fetzi**
add spatie/laravel-feed for RSS feed generation

fixes #339

### CODE REVIEW

**Comment by driesvints** [comment]
@fetzi can you perhaps instead of adding a merge commit from master, rebase the two commits with master? Trying to avoid merge commits if we can.

**Comment by driesvints** [comment]
Build was also not running but I noticed a setting wasn't properly set for that. Should be fixed now and the build should run on your next push.

**Comment by fetzi** [comment]
@driesvints I've rebased with master, but styleci/push did not trigger. 🤔

**Comment by driesvints** [comment]
Really odd. @GrahamCampbell any idea why the push hook didn't trigger?

**Comment by fetzi** [comment]
@driesvints any update?

**Comment by driesvints** [comment]
I'm really sorry this is taking so long. I'm having a hard time finding the time to review PR's for Laravel.io atm and I can't make any promises anymore when I'll get to it. I'm afraid I just have to say that I'll eventually take a look at it, hopefully sooner rather than later.

**Comment by GrahamCampbell** [comment]
> Really odd. @GrahamCampbell any idea why the push hook didn't trigger?

Push analyses don't run on forks.

**Comment by driesvints** [comment]
@GrahamCampbell thanks for explaining!

File: app/Models/Thread.php [hunk 1]
```diff
@@ -146,4 +164,15 @@ public static function feedQuery(): Builder
...
+
+    /**
+     * retrieve 20 feed items with use of page parameter.
+     */
+    public static function getFeedItems()
+    {
+        $page = intval(request('page', 1));
+        $query = static::feedQuery();
+
+        return $query->skip(($page - 1) * static::FEED_PAGE_SIZE)->take(static::FEED_PAGE_SIZE)->get();
```
Comment (kubk) [inline app/Models/Thread.php]: It is possible to use built-in Laravel pagination here?

https://github.com/laravel/framework/blob/5.6/src/Illuminate/Database/Eloquent/Builder.php#L738

**Comment by joedixon** [comment]
Hey @driesvints - gave this some love.

- Rebased on master
- Bumped spatie/laravel-feed
- Removed published views to rely on defaults

I left my commits for clarity so let me know if you want me to squash.

File: app/Models/Thread.php [hunk 2]
```diff
@@ -173,4 +191,15 @@ public static function resolutionTime()
             return false;
         }
     }
+
+    /**
+     * retrieve 20 feed items with use of page parameter.
+     */
```
Comment (driesvints) [inline app/Models/Thread.php]: Remove this docblock.

File: app/Models/Thread.php (hunk 1 above)
Comment (driesvints) [reply app/Models/Thread.php]: @joedixon is this possible? ^

File: app/Models/Thread.php [hunk 3]
```diff
@@ -109,6 +114,19 @@ public function delete()
         parent::delete();
     }
 
+    public function toFeedItem()
```
Comment (driesvints) [inline app/Models/Thread.php]: Add a return type.

File: app/Models/Thread.php [hunk 4]
```diff
@@ -173,4 +191,15 @@ public static function resolutionTime()
             return false;
         }
     }
+
+    /**
+     * retrieve 20 feed items with use of page parameter.
+     */
+    public static function getFeedItems()
```
Comment (driesvints) [inline app/Models/Thread.php]: Add a return type.

File: config/feed.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+             * ['App\Model@getAllFeedItems', 'argument']
+             */
+            'items' => ['App\Models\Thread@getFeedItems'],
+
+            /*
+             * The feed will be available on this url.
+             */
+            'url' => '/forum/rss',
+
+            'title' => 'laravel.io Forum RSS Feed',
```
Comment (driesvints) [inline config/feed.php]:
```suggestion
            'title' => 'Laravel.io Forum RSS Feed',
```

File: config/feed.php [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        'forum' => [
+            /*
+             * Here you can specify which class and method will return
+             * the items that should appear in the feed. For example:
+             * 'App\Model@getAllFeedItems'
+             *
+             * You can also pass an argument to that method:
+             * ['App\Model@getAllFeedItems', 'argument']
+             */
+            'items' => ['App\Models\Thread@getFeedItems'],
```
Comment (driesvints) [inline config/feed.php]:
```suggestion
            'items' => [App\Models\Thread::class . '@getFeedItems'],
```

**Comment by joedixon** [comment]
> How do people know where to find the RSS feed?

There is an atom link in the head of the page so people can find it with their RSS reader.

For something more explicit, perhaps a link in the footer or do you want it more prominent than that?

**Comment by driesvints** [comment]
@joedixon I was thinking a button below "create thread" on the forum overview page.

**Comment by joedixon** [comment]
All suggested changes made including using native pagination on the feed.

Also changed the feed URL from `/forum/rss` to `/forum/feed`.

File: config/database.php [hunk 7]
```diff
@@ -56,7 +56,7 @@
             'collation' => 'utf8mb4_unicode_ci',
             'prefix' => '',
             'prefix_indexes' => true,
-            'strict' => true,
+            'strict' => false,
```
Comment (driesvints) [inline config/database.php]: Why did you change this? Not sure if that's wanted for production?

**Comment by driesvints** [comment]
@joedixon can you also add the button?

File: config/database.php (hunk 7 above)
Comment (joedixon) [reply config/database.php]: Accidental commit. I have to do this to get it working locally. I will revert.

**Comment by joedixon** [comment]
Screenshot 2019-11-21 at 21 40 38

**Comment by driesvints** [comment]
Awesome. Thanks @fetzi & @joedixon!
